From 77c50aff8a2cde1532aa1090465581659cfbc6fc Mon Sep 17 00:00:00 2001
From: ywan171 <yi.a.wang@intel.com>
Date: Thu, 26 Jul 2012 08:00:06 +0800
Subject: NuPlayerRender: fix a/v sync issue when playback only video track clip

BZ: 48710

orig patch:
43398:fix google bugs
If only video track exists, mAnchorTimeMediaUs retrieve system time as a/v sync time base.
If playing is paused, we should clear mAnchorTimeMediaUsand reset it when resume playing.

Category: aosp improvement
Domain: Video.Media-streaming
Origin: internal
Upstream-Candidate: yes

Change-Id: I5801a791d3b1b89ea6bd0f07a3caeb3079860f47
Orig-Change-Id: If71b0e83b0bbf0d9f7687675e5b0375e7eff8595
Signed-off-by: ywan171 <yi.a.wang@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
---
 .../nuplayer/NuPlayerRenderer.cpp                  |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp b/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp
index 3b2784b..942633c 100644
--- a/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp
+++ b/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp
@@ -690,6 +690,9 @@ void NuPlayer::Renderer::onPause() {
 
     if (mHasAudio) {
         mAudioSink->pause();
+    } else {
+        // if no audio, we need to clear mAnchorTimeMediaUs, reset it when first video buffer comes
+        mAnchorTimeMediaUs = -1;
     }
 
     ALOGV("now paused audio queue has %d entries, video has %d entries",
-- 
1.7.4.1

